source/mir/random/engine/package.d

```diff
@@ -199,6 +270,282 @@ version(mir_random_test) unittest
     static assert(is(EngineReturnType!Random == size_t));
 }
 
+static if (THREAD_LOCAL_STORAGE_AVAILABLE)
+{
+    /++
+    Thread-local instance of the default $(LREF Random) allocated and seeded independently
+    for each thread. Requires $(LINK2 https://en.wikipedia.org/wiki/Thread-local_storage, TLS).
+    +/
+    alias rne = threadLocal!Random;
+    ///
+    @nogc nothrow @safe version(mir_random_test) unittest
+    {
+        import mir.random;
+        import std.complex;
+
+        auto c = complex(rne.rand!real, rne.rand!real);
+
+        //If you are going use rne several times in a row,
+        //it's more efficient to retain rne like this:
+        auto rng = rne;
```

Comment:
Maybe we should prohibit this functionality because is very unsafe? We can do it if we return TL!Engine by reference (its copy constructor already disabled)

Reply:
If one want best speed hi can init local Engine or use the PTR.